mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -896,6 +902,42 @@ impl S3RequestError {
     }
 }
 
+impl ProvideErrorMetadata for S3RequestError {
+    fn meta(&self) -> ClientErrorMetadata {
+        match self {
+            Self::ResponseError(request_result) => {
+                let http_code = if request_result.response_status >= 100 {
+                    Some(request_result.response_status)
+                } else {
+                    None
+                };
+                ClientErrorMetadata {
+                    http_code,
+                    ..Default::default()
+                }
+            }
+            Self::Forbidden(_, metadata) => (**metadata).clone(),
+            Self::Throttled => {
+                // CRT does not set S3 response data for error codes other then AWS_ERROR_S3_INVALID_RESPONSE_STATUS
```

Comment:
nit: not sure this comment is relevant anymore. Maybe some of it (the first two lines?) could make more sense in `try_parse_throttled`.

Reply:
Removed it for now. Not sure if this comment explains anything that happens in `try_parse_throttled`. I can add a comment to the match block, where we invoke parsers:
```rust
403 => try_parse_forbidden(request_result),
// CRT does not set S3 response status for error codes other then AWS_ERROR_S3_INVALID_RESPONSE_STATUS
0 => {
```